dataflow/src/node/process.rs

```diff
@@ -202,6 +202,40 @@ impl Node {
                     _ => {}
                 };
 
+                // When we miss at a can_query_through node we have to replay from
+                // its parent instead of that node, so we'll update the misses:
+                let reroute_miss = |internal: &NodeOperator, miss: &mut Miss| {
+                    if internal.can_query_through() {
+                        for col in miss.lookup_idx.iter_mut() {
+                            let parents = internal.resolve(*col).unwrap();
+                            assert_eq!(
+                                parents.len(),
+                                1,
+                                "can't query_through with more than one parent"
+                            );
+
+                            let (parent, parent_col) = parents[0];
+                            let parent_node = nodes
+                                .values()
+                                .find(|n| n.borrow().global_addr() == parent)
+                                .unwrap();
+                            miss.on = *parent_node.borrow().local_addr();
+                            *col = parent_col;
+                        }
+                    }
+                };
+
+                for miss in misses.iter_mut() {
+                    if miss.on == addr {
```

Comment:
Why is this condition here? I don't think there are any operators that do lookups on themselves and are *also* query through? At least for now.